spec/controllers/network_router_controller_spec.rb

```diff
@@ -180,6 +180,23 @@
     end
   end
 
+  describe "#delete_network_routers" do
+    let(:router) { FactoryGirl.create(:network_router, :name => "router-01") }
+    before do
+      controller.instance_variable_set(:@_params, :id => router.id)
+      controller.instance_variable_set(:@lastaction, "show")
+      controller.instance_variable_set(:@layout, "network_router")
+
+      allow_any_instance_of(NetworkRouter).to receive(:supports_delete?).and_return(true)
+      allow_any_instance_of(NetworkRouter).to receive(:delete_network_router_queue).and_return(true)
```

Comment:
Calling `allow_any_instance_of` is really expensive and 99% of the cases it can be avoided. I would rather mock the `find_by` [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/3796/files#diff-830cda38240c4d54e3ca344b49a1264dR166) to return with the given router and then you can simply mock the `router` object created by `FactoryGirl`.

```ruby
allow(NetworkRouter).to receive(:find_by).and_return(router)
allow(router).to receive(:supports_delete?).and_return(true)
allow(router).to receive(:delete_network_router_queue).and_return(true)
```